### PR TITLE
Adding data for persistence

### PR DESCRIPTION
**Pull request by josediegorobles**
I need that when lambda calls core methods they returns more data about previous utxo, new (for the sender) utxo and about the blinded utxo, without changing the callings from bitmask extension.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -561,7 +561,7 @@ pub async fn send_assets(
     info!("Creating transfer PSBT...");
 
     #[cfg(not(target_arch = "wasm32"))]
-    let (consignment, psbt, disclosure) = transfer_assets(
+    let (consignment, psbt, disclosure, _change, _previous_utxo, _new_utxo) = transfer_assets(
```
Comment (cryptoquick) [inline src/lib.rs]: Mind if we change these to a `TransferAssetsResponse` struct? Then we can just pass it around.

File: src/lib.rs [hunk 2]
```diff
@@ -669,12 +682,14 @@ pub async fn accept_transfer(
     blinding_factor: &str,
     outpoint: &str,
 ) -> Result<AcceptResponse> {
+    info!("ajaja");
```
Comment (cryptoquick) [inline src/lib.rs]: lol

File: src/web.rs [hunk 3]
```diff
@@ -268,7 +268,7 @@ pub fn send_assets(
 #[wasm_bindgen]
 pub fn accept_transfer(consignment: String, blinding_factor: String, outpoint: String) -> Promise {
     set_panic_hook();
-
+    crate::info!("aqui estamos");
```
Comment (cryptoquick) [inline src/web.rs]: Maybe set a more informative log message if you want to leave this in.

File: src/operations/rgb/send_tokens.rs [hunk 4]
```diff
@@ -191,7 +194,7 @@ pub async fn transfer_asset(
     let change_output = change_outputs.get(0).unwrap();
     debug!(format!("Selected change output: {change_output:#?}"));
 
-    let change = change
+    let change: BTreeMap<bp::seals::txout::blind::RevealedSeal, u64> = change
```
Comment (cryptoquick) [inline src/operations/rgb/send_tokens.rs]: It'd be better to use `RevealedSeal` from the imports section.

**Review by cryptoquick — CHANGES_REQUESTED** [review]
Just a couple of changes, but it looks good.

**Review by grunch — APPROVED** [review]
LGTM!

File: src/web.rs (hunk 3 above)
Comment (josediegorobles) [reply src/web.rs]: No, i left several debug logs that are unnecessary...

File: src/lib.rs (hunk 1 above)
Comment (josediegorobles) [reply src/lib.rs]: Yeah much better

**Comment by josediegorobles** [comment]
Now I'm calling a new endpoint, declare, that persist the data of the receptor and the change data of the sender. Now i'm sending all as soon as the transaction is sent.

File: src/lib.rs [hunk 5]
```diff
@@ -624,11 +633,14 @@ pub async fn transfer_assets(
     String, // bech32m compressed sten consignment
     String, // base64 bitcoin encoded psbt
     String, // json
+    String,
```
Comment (grunch) [inline src/lib.rs]: It would be great to comment those new strings, change, previous_utxo and new_utxo

File: src/operations/rgb/send_tokens.rs [hunk 6]
```diff
@@ -290,7 +286,7 @@ pub async fn transfer_asset(
                 "Error constructing PSBT from RGB Tokens Descriptor: {err}",
             ));
             return Err(anyhow!(
-                "Error constructing PSBT from RGB Tokens Descriptor"
+                "jaja: Error constructing PSBT from RGB Tokens Descriptor: {err}"
```
Comment (grunch) [inline src/operations/rgb/send_tokens.rs]: 🤣

**Review by grunch — CHANGES_REQUESTED** [review]

File: src/operations/rgb/send_tokens.rs (hunk 6 above)
Comment (josediegorobles) [reply src/operations/rgb/send_tokens.rs]: Haha.

**Review by grunch — APPROVED** [review]

**Review by cryptoquick — APPROVED** [review]
ACK. LGTM!

**Pull request merged**